addons/point_of_sale/tests/test_point_of_sale_flow.py

```diff
@@ -1064,3 +1064,64 @@ def test_order_refund_picking(self):
 
         refund.action_pos_order_invoice()
         self.assertEqual(refund.picking_count, 1)
+
+    def test_order_with_different_payments_and_refund(self):
+        """
+        Test that all the payments are correctly taken into account when the order contains multiple payments and money refund.
+        In this example, we create an order with two payments for a product of 750$:
+            - one payment of $300 with customer account
+            - one payment of $460 with cash
+        Then, we refund the order with $10, and check that the amount still due is 300$.
+        """
+
+        product5 = self.env['product.product'].create({
+            'name': 'product5',
+            'type': 'product',
+            'categ_id': self.env.ref('product.product_category_all').id,
+        })
+
+        # sell product thru pos
+        self.pos_config.open_session_cb()
+        pos_session = self.pos_config.current_session_id
+        product5_order = {'data':
+          {'amount_paid': 750,
+           'amount_return': 10,
+           'amount_tax': 0,
+           'amount_total': 750,
+           'creation_date': fields.Datetime.to_string(fields.Datetime.now()),
```

Comment:
If the date isn't really important (nothing using date seems to be tested here), just hard code it.

Reply:
Or leave it that way, it seems that there are a lot of `now()` in this file anyway